services/QuillLMS/engines/evidence/app/controllers/evidence/research/gen_ai/datasets_controller.rb

```diff
@@ -84,15 +84,33 @@ def annotation_tool_data
           @llms = Llm.all_ordered
           @llm_prompt_templates = LlmPromptTemplate.order(:order)
           @relevant_texts = @dataset.stem_vault.relevant_texts.includes(:dataset_relevant_texts)
+          @clusters = @dataset.clusters.ordered
 
-          all_examples = @test_examples + @prompt_examples
-          cluster_ids = all_examples.map(&:cluster_id).compact.uniq
-          @clusters = cluster_ids.any? ? Cluster.where(id: cluster_ids).ordered : []
+          render json: build_editor_data
+        end
+
+        def save_changes
+          dataset = Dataset.find(params[:id])
+          changes = DatasetEditor::BooleanParamsNormalizer.run(changes_params).to_h
+
+          result = DatasetEditor::ChangesSaver.run(dataset:, changes:)
 
-          render json: build_annotation_tool_data
+          if result[:success]
+            if result[:new_dataset_id]
+              render json: { success: true, new_dataset_id: result[:new_dataset_id], version_type: result[:version_type] }
+            else
+              render json: { success: true }
+            end
+          else
+            render json: { success: false, error: result[:error] }, status: :unprocessable_entity
+          end
+        rescue StandardError => e
+          Rails.logger.error "Error saving changes: #{e.message}"
```

Comment:
Probably should don't need this in prod. Maybe use `ErrorNotifier` if you need it.